addons/hr_org_chart/static/src/fields/hr_org_chart.js

```diff
@@ -48,25 +49,25 @@ export class HrOrgChart extends Component {
         this.lastParent = null;
         this._onEmployeeSubRedirect = onEmployeeSubRedirect();
 
-        onWillStart(this.handleComponentUpdate.bind(this));
-        onWillRender(this.handleComponentUpdate.bind(this));
-    }
+        onWillStart(async () => {
+            this.employee = this.props.record.data;
+            // the widget is either dispayed in the context of a hr.employee form or a res.users form
+            this.state.employee_id = this.employee.employee_ids !== undefined ? this.employee.employee_ids.resIds[0] : this.props.record.resId;
+            this.state.parent_id = this.employee.parent_id?.[0] || this.employee.employee_parent_id?.[0];
+            this.original_employee_id = this.state.employee_id;
+        });
 
-    /**
-     * Called on start and on render
-     */
-    async handleComponentUpdate() {
-        this.employee = this.props.record.data;
-        // the widget is either dispayed in the context of a hr.employee form or a res.users form
-        this.state.employee_id = this.employee.employee_ids !== undefined ? this.employee.employee_ids.resIds[0] : this.props.record.resId;
-        const manager = this.employee.parent_id || this.employee.employee_parent_id;
-        const forceReload = this.lastRecord !== this.props.record || this.lastParent != manager;
-        this.lastParent = manager;
-        this.lastRecord = this.props.record;
-        await this.fetchEmployeeData(this.state.employee_id, forceReload);
+        useRecordObserver(async (record) => {
+            this.state.employee_id = this.original_employee_id;
+            let new_parent_id = record.data.parent_id?.[0] || false;
+            if(this.lastParent !== new_parent_id){
```

Comment:
We could let the parenthesis breathe a bit. 😇 

```suggestion
            if (this.lastParent !== new_parent_id) {
```